facade/src/main/scala/reactST/reactTable/TableMaker.scala

```diff
@@ -0,0 +1,327 @@
+   *   should make use of the column instance "props" methods.
+   * The typescript definitions don't seem to include the react-table
+   *   support for footers. So, they aren't available here, either.
+   *   So, table footers are fully "Build It Yourself". This actually
+   *   has some advantages, because it is often useful for table footers
+   *   to span multiple columns and react-table does not support that.
+   *   At some point, something similar for an "extra" header row might
+   *   be useful since header groups have some issues.
+   */
+  def makeTable(
+    options:      TableOptsD,
+    data:         js.Array[D],
+    headerCellFn: Option[ColumnInstanceD => TagMod],
+    tableClass:   Css = Css(""),
+    footer:       TagMod = TagMod.empty
+  ) = {
+    val component = ScalaFnComponent[TableOptsD] { opts =>
+      val tableInstance = use(opts.setData(data))
+      val bodyProps     = tableInstance.getTableBodyProps()
+
+      val header = headerCellFn.fold(TagMod.empty) { f =>
+        <.thead(
+          tableInstance.headerGroups.toTagMod { g =>
+            <.tr(
+              TableMaker.props2Attrs(g.getHeaderGroupProps()),
+              headersFromGroup(g).toTagMod(f(_))
+            )
+          }
+        )
+      }
+
+      val rows = tableInstance.rows.toTagMod { rd =>
+        tableInstance.prepareRow(rd)
+        val cells = rd.cells.toTagMod { cell =>
+          <.td(TableMaker.props2Attrs(cell.getCellProps()), cell.render("Cell"))
+        }
+        <.tr(TableMaker.props2Attrs(rd.getRowProps()), cells)
+      }
+
+      <.table(tableClass, header, <.tbody(TableMaker.props2Attrs(bodyProps), rows), footer)
+    }
+    component(options)
+  }
+
+  /*
+   * When adding new plugs, see https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/react-table
+   * for help determining what the necessary type changes are.
+   *
+   * Heads Up! It doesn't seem to be documented, but the order of plugins seems
+   * to matter. If you put useSortBy before useFilters you will get a runtime error.
+   * We may want to add some means of ordering the plugs when we submit them to
+   * "useTable" if we can figure out what the required order is.
+   */
+
+  /**
+   * Add sort capabilities to the table via the useSortBy plugin hook.
+   */
+  // format: off
+  def withSort = new TableMaker[
+    D, 
+    TableOptsD with UseSortByOptions[D], 
+    TableInstanceD with UseSortByInstanceProps[D], 
+    ColumnOptsD with UseSortByColumnOptions[D], 
+    ColumnInstanceD with UseSortByColumnProps[D], 
+    State with UseSortByState[D]] {
+    val plugins = self.plugins :+ useSortBy.asInstanceOf[PluginHook[D]]
+  }
+  // format: on
+
+  // When trying to use a more traditional "syntax" package and implicit
+  // classes, the compiler seems to somehow loose type information along
+  // the way. Putting the syntax here allows resolution to work correctly.
+  object syntax {
```

Comment:
I added trailing underscores to the syntax methods to distinguish them from the "built in" ones since the the names clashed. I probably should just come up with some better names...